internal/app/rostermanagement.go

```diff
@@ -0,0 +1,106 @@
+package app
+
+import "github.com/TheTh1rt33nth/Nexus-octo-engine/pkg/model"
+
+func CreateRoster(user int, campaign int, name string) {
+	dummyCampaign := model.Campaign{
+		Name:       "Battle for Rum bottle",
+		StartingPL: 25,
+	}
+
+	// if /*check DBcontext if campaign already have a roster associated with tis name*/ {
+	/* fmt.Println("Go fuck yourself, you are already in this shit")
+	}
+	else
+	{ */
+	roster := model.Army{
+		Name:        name,
+		UserID:      user,
+		CampaignID:  campaign,
+		SupplyLimit: dummyCampaign.StartingPL, // ask context for ACTUAL campaign by id
+		SupplyUsed:  0,                        // tomodel
+		// send dis shit to DBcontext, we're done here
+	}
+}
+
+func CountCP(roster model.Army) int {
+	cp := 0
+	pl := 0
+	dummydetaches := []model.Detachment{}
+	if len(dummydetaches) == 0 {
+		return 0
+	}
+	for _, d := range dummydetaches {
+		cp -= d.CpCost
+		pl += d.PowerLevel
+	}
+	cp += CPbyPL(pl)
+	return cp
+}
+
+func CPbyPL(pl int) int {
+	switch {
+	case pl <= 50:
+		return 3
+	case pl <= 100:
+		return 6
+	case pl <= 200:
+		return 12
+	case pl > 200:
+		return 18
+	}
+
+	return 0
+}
+
+func DetachCost(dType string) int {
+	switch {
+	case dType == "Patrol":
```

Comment:
how about something like this:

a enum with detaches:
```go
type Detach string

const (
    Patrol Detach = "patrol"
    Batallion Detach = "batallion"
    // ...
)
```

And a map with costs:
```go
var DetachCosts map[Detach]int {
    Patrol: 2,
    Batallion: 3,
}
```

Reply:
"a enum".... wow so englisch